neqo-http3/src/connection.rs

```diff
@@ -3483,4 +3477,108 @@ mod tests {
             Err(Error::InvalidStreamId)
         );
     }
+
+    #[test]
+    fn test_read_frames_header_blocked() {
+        let (mut hconn, mut neqo_trans_conn, _, mut encoder) =
+            connect_and_receive_control_stream(true);
+        let request_stream_id = hconn
+            .fetch(
+                "GET",
+                "https",
+                "something.com",
+                "/",
+                &Vec::<(String, String)>::new(),
+            )
+            .unwrap();
+        assert_eq!(request_stream_id, 0);
+        let _ = hconn.stream_close_send(request_stream_id);
+        let out = hconn.process(None, now());
+        neqo_trans_conn.process(out.dgram(), now());
+
+        encoder.set_max_capacity(100).unwrap();
+        encoder.set_max_blocked_streams(100).unwrap();
+
+        // find the new request/response stream and send frame v on it.
+        let events = neqo_trans_conn.events();
+        for e in events {
+            match e {
+                ConnectionEvent::NewStream {
+                    stream_id,
+                    stream_type,
+                } => {
+                    assert_eq!(stream_id, request_stream_id);
+                    assert_eq!(stream_type, StreamType::BiDi);
+                }
+                ConnectionEvent::RecvStreamReadable { stream_id } => {
+                    assert_eq!(stream_id, request_stream_id);
```

Comment:
You can do this entire piece outside of the event loop.  Only the assertion needs to be here.

I say that because this whole blob could benefit from being extracted into a function that is called `send_response()` or similar.

Reply:
I will merge this patch and fix the test in issue #211